BitList.cs

```diff
@@ -0,0 +1,1552 @@
+        }
+        this.bits = new BitArray(_byteList.ToArray());
+    }
+
+    IEnumerator<bool> IEnumerable<bool>.GetEnumerator()
+    {
+        return new BitListEnumerator(this);
+    }
+
+    IEnumerator IEnumerable.GetEnumerator()
+    {
+        return new BitListEnumerator(this);
+    }
+
+    /// <summary>
+    /// Get the index of a bit in this instance.
+    /// </summary>
+    /// <param name="bit">A bit.</param>
+    /// <returns>The <paramref name="bit"/> index.</returns>
+    public int IndexOf(bool bit)
+    {
+        for (int i = 0; i < Count; i++)
+        {
+            if (bit == this[i])
+                return i;
+        }
+        return -1;
+    }
+
+    /// <summary>
+    /// Get the index of a list of bit in this instance.
+    /// </summary>
+    /// <param name="bits">A list of bit.</param>
+    public int IndexOf(BitList bits)
+    {
+        for (int i = 0; i < Count; i++)
+        {
+            if (bits == this[i, bits.Count])
+                return i;
+        }
+        return -1;
+    }
+
+    /// <summary>
+    /// Clone this instance in a new object.
+    /// </summary>
+    public object Clone()
+    {
+        return this;
+    }
+
+    /// <summary>
+    /// Return true if the specified BitList is equal to this istance.
+    /// </summary>
+    /// <param name="other">A list of bit.</param>
+    public bool Equals(BitList other)
+    {
+        return Equals(this, other);
+    }
+
+    /// <summary>
+    /// Return true if this instance contains the specified bit.
+    /// </summary>
+    /// <param name="bit">A bit.</param>
+    public bool Contains(bool bit)
+    {
+        return IndexOf(bit) != -1;
+    }
+
+    /// <summary>
+    /// Return true if this instance contains the specified list of bit.
+    /// </summary>
+    /// <param name="bits">A list of bit.</param>
+    public bool Contains(BitList bits)
+    {
+        return IndexOf(bits) != -1;
+    }
+
+    /// <summary>
+    /// Shift bits to left.
+    /// </summary>
+    public void LeftShift(int count)
+    {
+        if (count != 0)
+        {
+            for (int i = 1; i == Math.Abs(count); i++)
+            {
+                if (count > 0)
+                    Add(false);
+                else
+                {
+                    try
+                    {
+                        RemoveAt(Count - 1);
+                    }
+                    catch
+                    {
+                        break;
+                    }
+                }
+            }
+            if (Count == 0 || IsEmpty)
+                Add(false);
+        }
+    }
+
+    /// <summary>
+    /// Shift bits to right.
+    /// </summary>
+    public void RightShift(int count)
+    {
+        if (count != 0)
+        {
+            for (int i = 1; i == Math.Abs(count); i++)
+            {
+                if (count < 0)
+                    Add(false);
+                else
+                {
+                    try
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this code to not nest more than 3 control flow statements.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5545753)